pkg/universe.dagger.io/go/configure.cue

```diff
@@ -0,0 +1,63 @@
+package go
+
+import (
+	"dagger.io/dagger"
+	"dagger.io/dagger/core"
+	"universe.dagger.io/docker"
+)
+
+// Configure installs go into the given input image.
+// It sets GOROOT, GOPATH, and adds the relevent go bin paths to PATH.
+#Configure: {
+	input: docker.#Image
+	// path to install go to in `input` and set the GOROOT env var to
+	// In most cases you want to use the default value provided here.
+	goroot: string | *"/usr/local/go"
+	// path for GOPATH to be set to
+	// In most cases you want to use the default value provided here.
+	gopath: string | *"/go"
+
+	// filesystem with go in it.
+	// This will be copied into the image at the path specified by `goroot`
+	// if `source` is specified just the path to the source will be copied.
+	contents: dagger.#FS
+	// path to go root in the `contents` filesystem
+	source: string | *"/"
```

Comment:
we could use `core.#Exec` and `go env GOROOT` here to make it more robust. Having said that, since this is a `dagger.#FS` and we don't have a `PATH`, figuring out where the `go` binary is located, is not straightforward. 

Otherwise, how about defaulting this to `/usr/local/go` also since usually `GOROOT` is set to that?

Reply:
Agreed, probably most people will use this with a go image so defaulting to `/usr/local/go` is ideal.
Updated.

Reply:
To that end, I also added a `ref: core.#Ref` to have this action pull the image directly instead of having to pull outside the action.
LMK what you think.
I'm not sure I 100% get the usage of disjunctions or if it could possibly be done better here.